Added directory as parameter and used find instead of for

mateusmuller opened the pull request:
Added:
- Directory as a parameter
- Validation for empty parameters and if the directory exists
- Used "find" instead of "for"

Commit:
Added directory as parameter and used find instead of for

championballer reviewed (changes requested):
@mateusmuller Can you also add a conditional where the user can choose to either make a directory or a standalone file as exec? I guess in your fork, it replaces the file code with directory. Please correct me if I am wrong.

mateusmuller commented:
Hi @championballer!
Apologies, I don't get this part "replaces the file code with directory"...
My main idea was to pass both the file type and the directory where it should search of the files, so:

- $ find /tmp -iname *.py -exec chmod {} \\;

Where:
1. "/tmp" would be the DIRECTORY variable.
2. "py" would be the TYPE variable.

Yes, I will try to add those functionalities!

mateusmuller commented:
Hi @championballer ,
I believe it does not make much sense to have the options to turns either one file or directory as executable. This can simply be done using "chmod +x" and does not need a script for that.
My suggestion is to keep this script only for recursive "chmod +x" in a specific file type and a specific directory.

Commit:
Made DIRECTORY as requisite field. Throw error on stderr if not
informed.
FILE_EXTENSION variable, if not informed, will use "sh" (most common).

Commit:
Fixed spaces on my identation.
